ADBench/run-all.ps1

```diff
@@ -1,29 +1,109 @@
-function assert($expr)
-{
-    if (!(& $expr)) {
+# Accept command-line args
+param([int]$global:nruns_f=10, [int]$global:nruns_J=10, [string]$global:tmpdir="")
+
+# Assert function
+function assert ($expr) {
+    if (!(& $expr @args)) {
         throw "Assertion failed [$expr]"
     }
 }
 
-$dir = split-path ($MyInvocation.MyCommand.Path)
-assert { $dir -match 'ADbench$' }
-$dir = Split-Path $dir
+# Get source dir
+
+$global:dir = split-path ($MyInvocation.MyCommand.Path)
```

Comment:
Most of these should be not global: -- I can help figure out cases where that breaks something.

Reply:
It couldn't seem to access them from within the class methods otherwise - said something like 'variable not defined within method'

Reply:
oops, I submitted this comment too early... see fix below.